lib/srv/sess.go

```diff
@@ -338,6 +346,47 @@ func (s *SessionRegistry) GetTerminalSize(sessionID string) (*term.Winsize, erro
 	return sess.term.GetWinSize()
 }
 
+func (s *SessionRegistry) isApprovedFileTransfer(ctx context.Context, scx *ServerContext) (bool, error) {
```

Comment:
Let's add some tests for this function since it seems complex enough to warrn it.

Reply:
Gave the test a shot although the test seemed to be a bit complex (maybe unavoidable?). Happy for input